### PR TITLE
Golf proofs and fix typos

### PR DESCRIPTION
**Pull request by pitmonticone**
- [x] Golf proof of `periodicpts_is_mem`
- [x] Golf proof of `periodic_arbitrary_large_time`
- [x] Golf proof of `inter_subset_empty_of_inter_empty `
- [x] Golf proof of `separated_balls`
- [x] Golf proof of `separated_ball_image_ball`
- [x] Golf proof of `separated_balls_along_non_periodic_orbit` 
- [x] Golf proof of `ball_non_periodic_arbitrary_large_time`
- [x] Golf proof of `non_periodic_arbitrary_large_time`
- [x] Golf proof of `arbitrary_large_time`
- [x] Golf proof of `is_closed`
- [x] Golf proof of `is_cpt`
- [x] Golf proof of `omegaLimit_nonempty`
- [x] Golf proof of `omegaLimit_nonwandering`  
- [x] Golf proof of `nonWandering_nonempty`
- [x] Golf proof of `recurrentSet_iff_accumulation_point`
- [x] Golf proof of `periodicpts_mem_recurrentSet`
- [x] Golf proof of `recurrentSet_nonwandering`
- [x] Golf proof of `recurrentSet_of_minimal_is_all_space`
- [x] Golf proof of `minimalSubset_mem_recurrentSet`

### CODE REVIEW

File: BET/Topological.lean [hunk 1]
```diff
@@ -35,81 +35,54 @@ def nonWanderingSet (f : α → α) : Set α :=
...
-
 /-- The set of points which are not periodic of any period. -/
 def IsNotPeriodicPt (f : α → α)  (x : α) := ∀ n : ℕ, 0 < n -> ¬IsPeriodicPt f n x
 
-lemma separated_ball_image_ball (n : ℕ) (hn : 0 < n) (x : α) (hfx : IsNotPeriodicPt f x) :  ∃ (ε : ℝ), 0 < ε ∧ (ball x ε) ∩ (f^[n] '' (ball x ε)) = ∅ := by
-    have hfx2 := hfx n hn
-    have hfnC : Continuous f^[n] := Continuous.iterate hf n
-    have hfx2' : x ≠ f^[n] x := Ne.symm hfx2
-    exact separated_balls (f^[n]) hfnC x hfx2'
+lemma separated_ball_image_ball (n : ℕ) (hn : 0 < n) (x : α) (hfx : IsNotPeriodicPt f x) :
```
Comment (mseri) [inline BET/Topological.lean:80]: Is there a particular reason to avoid using `have` or intros?

Comment (pitmonticone) [reply BET/Topological.lean:80]: Not really. It's just that in this case we could shorten the proof by removing them.

**Comment by pitmonticone** [comment]
@mseri We have finished for now. It's ready to be merged if you like.

File: BET/Topological.lean [hunk 2]
```diff
@@ -35,476 +35,292 @@ def nonWanderingSet (f : α → α) : Set α :=
...
+    obtain ⟨N, k3⟩ := ulim (ball x (ε / 2)) (mem_ball_self e2pos) isOpen_ball
+    exact ⟨u N, k3 N (Nat.le_refl N), hu N⟩
+  simp only [nonWanderingSet, mem_ball, ne_eq, exists_and_left, mem_setOf_eq] at h4
+  -- let l1 := h4 (ε / 2) e2pos
+  -- rcases l1 with ⟨y, l1, ⟨n, l2, l3⟩⟩
+  -- obtain below is equivalent to the above two lines
+  obtain ⟨y, l1, ⟨n, l2, l3⟩⟩ := h4 (ε / 2) e2pos
+  obtain ⟨h6, h7, h8⟩ : y ∈ ball z (ε / 2) ∧ f^[n] y ∈ ball z (ε / 2) ∧ n ≠ 0 := ⟨l1, l2, l3⟩
+    -- use y, n -- note `use y, n` which is the same as `use y` and `use n`
+    -- simp -- was repeatedly doing `mem_ball.mp: y ∈ ball x ε → dist y x < ε `
```
Comment (mseri) [inline BET/Topological.lean]:
```suggestion
```
These seem no longer necessary

File: BET/Topological.lean [hunk 3]
```diff
@@ -35,476 +35,292 @@ def nonWanderingSet (f : α → α) : Set α :=
...
+  have m1 : dist y z + dist z x < ε := by
+    rw [mem_ball] at h3 h6
+    linarith
+  have h9 : y ∈ ball x ε := lt_of_le_of_lt (dist_triangle _ _ _) m1
+  -- simp -- was doing `mem_ball.mp: y ∈ ball x ε → dist y x < ε `
+  -- have : dist y x ≤ dist y z + dist z x := dist_triangle _ _ _
```
Comment (mseri) [inline BET/Topological.lean]:
```suggestion
```
I think also these. We can keep the part below for our understanding, but probably that will go as well at some point

File: BET/Topological.lean [hunk 4]
```diff
@@ -35,476 +35,292 @@ def nonWanderingSet (f : α → α) : Set α :=
...
+  -- Question: Why can I omit argument, but I can't in the line below?
+  -- Answer: In this case the argument can easily be inferred from the goal since you want
+  -- to prove that `dist y x ≤ dist y z + dist z x` and you are giving it a lemma that says
+  -- exactly that, so the only thing left to determine is what are the `x`, `y` and `z` in the
+  -- lemma. On the other hand in the line below the arguments are not obvious from the goal,
+  -- since you need to prove an inequality using some kind of transitivity, so it is easy to
+  -- infer the first and last objects from the goal, but not the middle one.
+  -- exact lt_of_le_of_lt this m1
```
Comment (mseri) [inline BET/Topological.lean]:
```suggestion
```

File: BET/Topological.lean [hunk 5]
```diff
@@ -35,476 +35,292 @@ def nonWanderingSet (f : α → α) : Set α :=
...
+  have p1 : dist (f^[n] y) z + dist z x < ε := by
+    rw [mem_ball] at h7 h3
+    linarith
+  have h10 : f^[n] y ∈ ball x ε := lt_of_le_of_lt (dist_triangle _ _ _) p1
+    -- simp -- was doing `mem_ball.mp: y ∈ ball x ε → dist y x < ε `
+    -- have : dist (f^[n] y) x ≤ dist (f^[n] y) z + dist z x := dist_triangle _ _ _
```
Comment (mseri) [inline BET/Topological.lean]:
```suggestion
```

File: BET/Topological.lean [hunk 6]
```diff
@@ -35,476 +35,292 @@ def nonWanderingSet (f : α → α) : Set α :=
...
+    rw [recurrentSet, mem_setOf_eq, mem_omegaLimit_iff_frequently] at recur_x
     have recur_x_in_ball := recur_x (ball x ε) (ball_mem_nhds x hε)
-    simp [frequently_atTop] at recur_x_in_ball
+    simp only [singleton_inter_nonempty, frequently_atTop] at recur_x_in_ball
     exact recur_x_in_ball N
   . intro hf
-    unfold recurrentSet
-    simp only [mem_setOf_eq] -- `x ∈ { y | p y } = p x`
-    rw [mem_omegaLimit_iff_frequently]
+    -- simp only [mem_setOf_eq] -- `x ∈ { y | p y } = p x`
```
Comment (mseri) [inline BET/Topological.lean]:
```suggestion
```

**Comment by mseri** [comment]
Thanks for the PR! We looked at it with @oliver-butterley and we think it is time to merge :)

**Comment by oliver-butterley** [comment]
Thanks @pitmonticone and @LorenzoLuccioli!

**Pull request merged**